docs: Getting started rewrite

Jayclifford345 opened the pull request:
**What this PR does / why we need it**:
With the focus change to two deployment modes, Monolthic and Microservice, it was time to refresh the quick start guide to Loki. This incorporates much of the same flavour as the original:
- Docker logs scraping
- LogQL examples
- Datasource setup

But includes newer features:
- Grafana Logs Drilldown
- Alloy Live debugging

Aswell as adding the Carnivouse Greehouse example to provide an interactive method to understanding LogQL queries.

Killercoda sample found here: https://killercoda.com/grafana-dev-testing/course/workshops/course-tracker-test 

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

Jayclifford345 commented:
@JStickler, for some reason, the tests keep failing on this one. We are ready for review, happy to discuss these changes on our call together as it was quite a big change to the getting started guide.

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +199,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
-      In the previous query, the label selector is `{container="evaluate-loki-flog-1"}`.
+      In the previous query, the label selector is `{container="greenhouse-main_app-1"}`.
 
-   1. To view all the log lines which have the container label `evaluate-loki-grafana-1`:
```
I'm sad that we've lost these other examples in the revision.  I was trying to build in some simple LogQL lessons with them.

Jayclifford345 replied:
hmm, let me work on that. I wonder if we can still achieve the same with the containers at our disposal.

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +199,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+<!-- INTERACTIVE page step5.md START -->
 
-For a thorough introduction to LogQL, refer to the [LogQL reference](https://grafana.com/docs/loki/<LOKI_VERSION>/query/).
+### Extracting Attributes from Logs
```
```suggestion
### Extracting attributes from logs
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +199,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+When you now expand a log line in the query result, you will see the extracted attributes.
+
+{{< admonition type="tip" >}}
+**Before we move on** to the next section, let's generate some error logs. To do this, enable the bug service in the sample application. This is done by setting the `Toggle Error Mode` to `On` in the Carnivorous Greenhouse application. This will cause the bug service to randomly cause services to fail.
```
I don't think we need the tip or bold formatting here.

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +199,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+`service_name` is a label created by Loki when no service name is provided in the log line. It will use the container name as the service name. A list of all labels can be found in [Labels](https://grafana.com/docs/loki/latest/get-started/labels/#default-labels-for-all-users).
+{{< /admonition >}}
+
+Lastly, lets take a look at the total log throughput of each container in our production environment:
```
```suggestion
Finally, lets take a look at the total log throughput of each container in our production environment:
```

JStickler reviewed (commented):
Only made it through the docs before I ran out of time.  I'll look at the KillerCoda tomorrow.

Jayclifford345 commented:
Aaah thanks @JStickler! All changes are accepted, and the killercoda has been updated for your testing.

JStickler reviewed (commented):
Made it through both the docs/local install and the Killercoda today.

Jayclifford345 commented:
Hey @JStickler, just accepted your amendments. Will work through the rest manually :)

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+* **Provisioning**: This is where you define the datasource in a configuration file and have Grafana automatically create the datasource.
+* **API**: This is where you use the Grafana API to create the datasource.
+
+In this case we are using the provisioning method. Instead of mounting the Grafana configuration directory, we have defined the datasource in the `docker-compose.yml` file:
```
```suggestion
In this case we are using the provisioning method. Instead of mounting the Grafana configuration directory, we have defined the data source in this portion of the `docker-compose.yml` file:
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+## A look under the hood
+
+At this point you will have a running Loki Stack and a sample application generating logs. You have also queried Loki using Grafana Logs Drilldown and Grafana Explore. 
+In this next section we will take a look under the hood to understand how the Loki Stack has been configured to collect logs, the Loki configuration file, and how the Loki datasource has been configured in Grafana.
```
```suggestion
In this next section we will take a look under the hood to understand how the Loki Stack has been configured to collect logs, the Loki configuration file, and how the Loki data source has been configured in Grafana.
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+
+<!-- INTERACTIVE page step8.md START -->
+
+### Grafana Loki Datasource
```
Global, data source is standardized as two words.
https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#data-source
```suggestion
### Grafana Loki Data source
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
 {{< /docs/ignore >}}
 
+You have completed the Loki Quickstart demo. So where to go next? Here are a few suggestions:
+* **Deploy:** Loki can be deployed in multiple ways. For production usecases we recommend deploying Loki via the [Helm chart](https://grafana.com/docs/loki/<LOKI_VERSION>/setup/install/helm/).
```
```suggestion
* **Deploy:** Loki can be deployed in multiple ways. For production use cases we recommend deploying Loki via the [Helm chart](https://grafana.com/docs/loki/<LOKI_VERSION>/setup/install/helm/).
```

JStickler reviewed (commented):
[docs team]  Took another run through this on my local today.  I didn't go through the Killercoda again because I'm not sure if that environment has been refreshed recently?

Jayclifford345 commented:
Hey @JStickler, that should be the Killercoda synced and the changes made: https://killercoda.com/grafana-dev-testing/course/workshops/course-tracker-test

Let me know your thoughts on the missing queries in this version of the getting started as I don't want us to lose value in the process of remaking this. Is it more the continuity of the example? :)

JStickler commented on docs/sources/get-started/quick-start.md:
````diff
@@ -64,124 +49,137 @@ It's a fully configured environment with all the dependencies already installed.
+      After running the command, you should see a similar output:
+
+      ```console
+       ✔ Container loki-fundamentals-grafana-1  Started  0.3s 
````
I'm not seeing the Drilldown menu item in Killercoda. Drilldown is still under Explore > Logs.  And the direct link at line 103 gives a 404.

And when I run it on my local, I'm getting the same result.  We've somehow lost the Drilldown menu that we had the last time I ran through this.

JStickler commented on docs/sources/get-started/quick-start.md:
````diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
 ```
 <!-- INTERACTIVE copy END -->
-The flog app generates log lines for simulated HTTP requests.
+When you now expand a log line in the query result, you will see the extracted attributes.
````
Since this is a getting started, let's not assume that users know about log details yet.
```suggestion
When you click the arrow (>) next to a log line in the query result to expand the view, you will see the extracted attributes in the log details.
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
-The flog app generates log lines for simulated HTTP requests.
+When you now expand a log line in the query result, you will see the extracted attributes.
+
+{{< admonition type="tip" >}}
```
I don't know if we want to move this tip higher up the page to give the service more time to generate errors, or add a line about waiting a minute or two, but a couple of times when I've run the first query, I haven't had any data that matched the query.  When I run it again, there's usually a result, but sometimes it takes a minute or two to generate enough errors to get data from my query.

JStickler commented on docs/sources/get-started/quick-start.md:
````diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
-```bash
-{container="evaluate-loki-flog-1"} | json | status="401"
+```logql
+sum(rate({container="greenhouse-plant_service-1"} | logfmt | level="ERROR" [$__auto]))
````
The data from this query is hard to see in the graph when it's set to `lines` since it's just a couple of green dots.  It might be helpful to add another sentence here and direct users to change the graph view to `bars` to make the data more visible in the graph.  That would also teach new users that there are multiple graph visualizations available that they can play around with.

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+
+### Grafana Alloy configuration
+
+Grafana Alloy is collecting logs from all the docker containers and forwarding them to Loki. 
```
```suggestion
Grafana Alloy is collecting logs from all the Docker containers and forwarding them to Loki. 
```

JStickler commented on docs/sources/get-started/quick-start.md:
````diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+It needs a configuration file to know which logs to collect and where to forward them to. Within the `loki-fundamentals` directory, you will find a file called `config.alloy`:
+
+```alloy
+// This component is responsible for disovering new containers within the docker environment
````
```suggestion
// This component is responsible for discovering new containers within the Docker environment
```

JStickler commented on docs/sources/get-started/quick-start.md:
```diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
 
-## Loki data source in Grafana
+In this view you can see the components of the Alloy configuration file and how they are connected:
+* **discovery.docker**: This component queries the metadata of the docker environment via the docker socket and discovers new containers, as well as providing metadata about the containers.
```
```suggestion
* **discovery.docker**: This component queries the metadata of the Docker environment via the Docker socket and discovers new containers, as well as providing metadata about the containers.
```

JStickler commented on docs/sources/get-started/quick-start.md:
````diff
@@ -194,165 +192,337 @@ You can view your logs using the command line interface, [LogCLI](/docs/loki/<LO
+  retention_enabled: true
+```
+To summarize the configuration file:
+* **auth_enabled**: This is set to false, meaning Loki does not need a [tenant ID](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/multi-tenancy/) for ingest or query. Note that this is not recommended for production environments. When deploying the Loki helm chart, this is set to true by default.
````
```suggestion
* **auth_enabled**: This is set to false, meaning Loki does not need a [tenant ID](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/multi-tenancy/) for ingest or query. Note that this is not recommended for production environments. When deploying the Loki Helm chart, this is set to true by default.
```

JStickler reviewed (commented):
[docs team]  Something's still off, I'm getting the Explore menu instead of the new Drilldown menu.  =(

Jayclifford345 commented:
Hey @JStickler, so i think i caught all the changes you made before we moved the tutorial so hopefully I caught them all. I have also fixed the Killercoda to the right URL and updated the Grafana version which should fix the old explore UI menu to Drill down. You just might need to do a full docker compose down and then up again to see it:
https://killercoda.com/grafana-dev-testing/course/loki/loki-getting-started-tutorial

JStickler reviewed (commented):
[docs team]  I tried running through the killercoda, but it's not working.  Maybe it's because the PR is so old, and we've had a new release?

Jayclifford345 commented:
That should be us up and running: https://killercoda.com/grafana-dev-testing/course/loki/loki-getting-started-tutorial with the latest changes :)

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
```diff
@@ -0,0 +1,519 @@
+
+Grafana Alloy is collecting logs from all the Docker containers and forwarding them to Loki. 
+It needs a configuration file to know which logs to collect and where to forward them to. Within the `loki-fundamentals` directory, you will find a file called `config.alloy`:
+
```
Do we have some kind of `docs ignore` command in the transformations?  Because I think we need to tell the user to click the *Editor* tab in the Killercoda environment before we ask them to look at the `config.alloy` file.

Jayclifford345 replied:
aaaah really good point

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
````diff
@@ -0,0 +1,519 @@
+  retention_enabled: true
+```
+To summarize the configuration file:
+* **auth_enabled**: This is set to false, meaning Loki does not need a [tenant ID](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/multi-tenancy/) for ingest or query. Note that this is not recommended for production environments. When deploying the Loki Helm chart, this is set to true by default.
````
```suggestion
* **auth_enabled**: This is set to `false`, meaning Loki does not need a [tenant ID](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/multi-tenancy/) for ingest or query. Note that this is not recommended for production environments. When deploying the Loki Helm chart, this is set to `true` by default.
```

JStickler reviewed (commented):
[docs team]  Yay, finally got the docker environment fixed, and I see Drilldown Logs, not Explore Logs.

Jayclifford345 commented:
@JStickler thats the note added :)
https://killercoda.com/grafana-dev-testing/course/loki/loki-getting-started-tutorial
<img width="585" alt="Screenshot 2025-05-01 at 11 48 15" src="https://github.com/user-attachments/assets/f50373b4-90c5-4439-80ca-07eca57618fb" />

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
```diff
@@ -0,0 +1,519 @@
+* **pattern_ingester**: Enables pattern ingesters which are used to discover log patterns. Mostly used by Grafana Logs Drilldown.
+* **ruler**: Enables the ruler component of Loki. This is used to create alerts based on log queries.
+* **frontend**: Defines the encoding format for the frontend. In this case it is set to `protobuf`.
+* **compactor**: Defines the compactor configuration. Used to compact the index and mange chunk retention.
```
```suggestion
* **compactor**: Defines the compactor configuration. Used to compact the index and manage chunk retention.
```

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
```diff
@@ -0,0 +1,519 @@
+
+### Grafana Loki Data source
+
+The final piece of the puzzle is the Grafana Loki datasource. This is used by Grafana to connect to Loki and query the logs. Grafana has multiple ways to define a datasource;
```
```suggestion
The final piece of the puzzle is the Grafana Loki data source. This is used by Grafana to connect to Loki and query the logs. Grafana has multiple ways to define a data source;
```

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
```diff
@@ -0,0 +1,519 @@
+### Grafana Loki Data source
+
+The final piece of the puzzle is the Grafana Loki datasource. This is used by Grafana to connect to Loki and query the logs. Grafana has multiple ways to define a datasource;
+* **Direct**: This is where you define the datasource in the Grafana UI.
+* **Provisioning**: This is where you define the datasource in a configuration file and have Grafana automatically create the datasource.
+* **API**: This is where you use the Grafana API to create the datasource.
```
```suggestion
* **Direct**: This is where you define the data source in the Grafana UI.
* **Provisioning**: This is where you define the data source in a configuration file and have Grafana automatically create the data source.
* **API**: This is where you use the Grafana API to create the data source.
```

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
````diff
@@ -0,0 +1,519 @@
+    networks:
+      - loki
+```
+Within the entrypoint section of the `docker-compose.yml` file, we have defined a file called `run.sh` this runs on startup and creates the datasource configuration file `ds.yaml` in the Grafana provisioning directory. 
````
```suggestion
Within the entrypoint section of the `docker-compose.yml` file, we have defined a file called `run.sh` this runs on startup and creates the data source configuration file `ds.yaml` in the Grafana provisioning directory. 
```

JStickler commented on docs/sources/get-started/quick-start/tutorial.md:
````diff
@@ -0,0 +1,519 @@
+      - loki
+```
+Within the entrypoint section of the `docker-compose.yml` file, we have defined a file called `run.sh` this runs on startup and creates the datasource configuration file `ds.yaml` in the Grafana provisioning directory. 
+This file defines the Loki datasource and tells Grafana to use it. Since Loki is running in the same Docker network as Grafana, we can use the service name `loki` as the URL.
````
```suggestion
This file defines the Loki data source and tells Grafana to use it. Since Loki is running in the same Docker network as Grafana, we can use the service name `loki` as the URL.
```

JStickler reviewed (commented):
[docs team]  Ran through the Killercoda today, everything looked good. I'm going to merge a few more typos I found and approve this PR.  But I want to wait to merge this until the engineering team finished backporting all the CI update from this week.  So look for this to merge late Friday or possibly Monday.  Yay, we finally got this one over the finish line!

JStickler reviewed (approved):
[docs team]
